File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        unordered_set<int> num_set(nums.begin(), nums.end());

        int max_sol = 1;
        for (int n : num_set) {
            if (!num_set.contains(n-1)) {
                int start = n;
                int sol = 1;
                while (num_set.contains(start+1)) {
                    start++;
                    sol++;
                }
                max_sol = max(max_sol, sol);
            }
        }
        return max_sol;
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.cs
public class Solution {
    public int MaxProduct(int[] nums) {
        int sol = nums.Max();
        int currMax = 1, currMin = 1;

        foreach (int n in nums) {
            if (n == 0) {
                currMax = 1;
                currMin = 1;
                continue;
            }

            int temp = currMax * n;
            currMax = Math.Max(temp, Math.Max(currMin * n, n)); 
            currMin = Math.Min(temp, Math.Min(currMin * n, n)); 
            sol = Math.Max(sol, currMax);
        }

        return sol;
    }
}
